src/adapters/chineseloc/chinese-source/chinese-source.js

```diff
@@ -0,0 +1,385 @@
+      let character = rawElement[0]
+      let typeCh = rawElement[1]
+      let typeText = rawElement[2]
+
+      let element = {
+        character
+      }
+      element[typeCh] = typeText
+
+      let prevElement = formattedData.get(rawElement[0])
+
+      if (prevElement) {
+        Object.keys(prevElement).filter(key => key !== 'character').forEach(key => {
+          element[key] = prevElement[key]
+        })
+      }
+      formattedData.set(character, element)
+    })
+    return formattedData
+    // console.info('formattedData - ', formattedData)
+  }
+
+  static convertRadicals (rawData) {
+
+  }
+
+  static findWord (targetWord, wordIDX, wordDict) {
+    let searchedIdxElement = wordIDX.get(targetWord)
+
+    if (searchedIdxElement) {
+      // console.info('serchedIdxElement.codes - ', searchedIdxElement.codes)
+      if (searchedIdxElement.codes && searchedIdxElement.codes.length > 0) {
+        // console.info('serchedIdxElement.codes - ', searchedIdxElement.codes)
+        return searchedIdxElement.codes.map(code => {
+          // console.info('findWord code - ', code, wordDict.get(code))
+          return wordDict.get(code)
+        })
+      }
+    }
+
+    return null
+  }
+
+  static lookupChinese (targetWord) {
+    // console.info('lookupChinese dHanziDict - ', dHanziDict.size)
+    let cpWord = targetWord
+    let count = 0
+    let format = 'simp'
+
+    let rs = []
+
+    let result
+    while (cpWord.length > 1) {
+      // console.info('cpWord - ', cpWord)
+      if (format === 'simp') {
+        result = ChineseSource.findWord(cpWord, dWordIndexSimp, dWordDict)
+        if (!result) {
+          format = 'trad'
+          result = ChineseSource.findWord(cpWord, dWordIndexTrad, dWordDict)
+        }
+      } else {
+        result = ChineseSource.findWord(cpWord, dWordIndexTrad, dWordDict)
+        if (!result) {
+          format = 'simp'
+          result = ChineseSource.findWord(cpWord, dWordIndexSimp, dWordDict)
+        }
+      }
+
+      if (result) {
+        result.forEach(resItem => {
+          rs[count++] = resItem
+          rs[count - 1].format = format
+
+          ChineseSource.formatDictionaryEntry(rs[count - 1])
+          ChineseSource.formatCharacterInfo(rs[count - 1], dHanziDict)
+          rs[count - 1].pinyin = ChineseSource.formatPinyin(rs[count - 1].pinyin)
+        })
+      }
+      cpWord = cpWord.substring(0, cpWord.length - 1)
+    }
+
+    return rs
+  }
+
+  static formatDictionaryEntry (resItem) {
+    if (resItem.format === 'trad') {
+      resItem.dictEntry = resItem.word.split(' ')[0]
+    } else if (resItem.format === 'simp') {
+      resItem.dictEntry = resItem.word.split(' ')[1]
+    }
+  }
+
+  static formatCharacterInfo (resItem) {
+    // console.info('formatCharacterInfo dHanziDict - ', dHanziDict)
+    const freqName =
+      [
+        'least frequent',
+        'less frequent',
+        'moderately frequent',
+        'more frequent',
+        'most frequent'

```

Comment:
I think it is from the discussion - where should we format the source data.
I think that as we get frequency feature from other sources in already formatted way.
Then it would not be consistent to format it on the clientAdapter side only for chinese.
@balmas , what do you think?

Reply:
Ok, I see your point @irina060981 . let's leave it for the moment.